federation/schema_syncer_test.go

```diff
@@ -351,3 +347,248 @@ func TestOnlyShadowServiceKnowsAboutNewField(t *testing.T) {
 	runAndValidateQueryResults(t, ctx, e, query, expectedOutput)
 
 }
+
+func createSchemasWithFederatedUser(t *testing.T) (*schemabuilder.Schema, *schemabuilder.Schema, *Executor) {
+	type User struct {
+		Id    int64
+		OrgId int64 `graphql:",optional"`
+	}
+	s1 := schemabuilder.NewSchemaWithName("schema1")
+	user := s1.Object("User", User{}, schemabuilder.RootObject)
+	user.Key("id")
+	s1.Query().FieldFunc("users", func(ctx context.Context) ([]*User, error) {
+		users := make([]*User, 0, 1)
+		users = append(users, &User{Id: int64(1), OrgId: int64(1)})
+		users = append(users, &User{Id: int64(2), OrgId: int64(2)})
+		return users, nil
+	})
+
+	type UserWithContactInfo struct {
+		Id    int64
+		OrgId int64
```

Comment:
If we change the OrgId to `optional` before removing it, can we avoid making the change in the previous commit?

Reply:
This is the case where we send it the org id, but that's not an arg its expected. The optional tag works the other way where we don't send it the org id, but its expecting it.

Reply:
Ah i see, thanks for the explanation!